src/evm/protocol/vm/adapter_contract.rs

```diff
@@ -51,12 +52,12 @@ where
         buy_token: Address,
         amounts: Vec<U256>,
         block: u64,
-        overwrites: Option<HashMap<rAddress, Overwrites>>,
+        overwrites: Option<HashMap<Address, Overwrites>>,
     ) -> Result<Vec<f64>, SimulationError> {
         let args = vec![
             Token::FixedBytes(pair_id),
-            Token::Address(sell_token),
-            Token::Address(buy_token),
+            Token::Address(H160(sell_token.0 .0)),
```

Comment:
Can't we create Address instead of going through H160? This keeps the `ethers` dependency that we're trying to get rid of

Reply:
yes I know.. I'm trying to do these in the smallest parts I can so it's not crazy overwhelming. 😓 
This part will be addressed in the last point of the TODOs in the description
> Use alloy to encode and decode instead of ethers (in simulation contract)